.github/workflows/ci.yml

```diff
@@ -64,6 +64,9 @@ jobs:
             --cache-from "$CORE_BRANCH_IMAGE:$BRANCH_REF" \
             .
       - name: Push dependabot-core-branch image to GHCR
+        env:
```

Comment:
This seems like a very sensible approach. In addition to protecting the GHCR, this should surface cases of misuse ✨

I've done a little googling, but I'd like to hear your thoughts on the name `ACCESS_CANARY`. My assumption is that it relates to pairing failed workflow runs to users. Is that correct?

Reply:
I'm not concerned about tracking the failures as that's currently the expectation for any PRs from external contributors or Dependabot. The GHCR push steps aren't required for those PRs so this is more about letting the rest of the workflow run so we can see if the tests are passing. Here's an example of how they are currently failing: https://github.com/dependabot/dependabot-core/runs/4644267200?check_suite_focus=true (sorry, should have linked that to start!)

I chose canary since it's an indirect indicator of whether the workflow will have the appropriate access to do the GHCR push. Sort of like the https://en.wiktionary.org/wiki/canary_in_a_coal_mine. I'd be open changing the name to something better but I'd like to merge this first to make sure it works.

Reply:
🚢🇮🇹

Reply:
Confirmed this worked! CI is passing now in this fork PR and we can see the GHCR push steps were skipped: https://github.com/dependabot/dependabot-core/runs/4653634875?check_suite_focus=true

Another option would have been to try and detect forks/dependabot directly like:
`if: github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]'` ([source](https://github.community/t/how-to-detect-a-pull-request-from-a-fork/18363/4))

I was worried there were more cases we'd need to detect and deal with so the secret check seemed like the simpler way to go. That alternative would make it more a little more clear which cases we expect to skip doing the GHCR push in.